packages/nextalign/src/nextalign.cpp

```diff
@@ -40,24 +43,33 @@ NextalignResultInternal nextalignInternal(const NucleotideSequence& query, const
   const auto gapOpenCloseNuc = getGapOpenCloseScoresCodonAware(ref, geneMap, options);
   const auto gapOpenCloseAA = getGapOpenCloseScoresFlat(ref, options);
 
+  debug_trace("Aligning nucleotide sequence\n", "");
   const auto alignmentStatus = alignPairwise(query, ref, gapOpenCloseNuc, options.alignment, options.seedNuc);
   if (alignmentStatus.status != Status::Success) {
     throw ErrorNonFatal(*alignmentStatus.error);
   }
 
+  const auto stripped = stripInsertions(alignmentStatus.result->ref, alignmentStatus.result->query);
+  const auto refStripped = removeGaps(ref);
+
   std::vector<PeptideInternal> queryPeptides;
   Warnings warnings;
   if (!geneMap.empty()) {
+    const auto querySize = safe_cast<int>(alignmentStatus.result->query.size());
+    const auto queryStrippedSize = safe_cast<int>(stripped.queryStripped.size());
+    const auto refSize = safe_cast<int>(alignmentStatus.result->ref.size());
+    const auto refStrippedSize = safe_cast<int>(refStripped.size());
+
+    const int bandWidth = safe_cast<int>(std::max(querySize - queryStrippedSize, refSize - refStrippedSize) / 3 + 3);
+    const int shift = bandWidth / 2;
+
     auto peptidesInternal = translateGenes(alignmentStatus.result->query, alignmentStatus.result->ref, refPeptides,
-      geneMap, gapOpenCloseAA, options);
+      geneMap, gapOpenCloseAA, options, bandWidth, shift);
     concat_move(peptidesInternal.queryPeptides, queryPeptides);
```

Comment:
This is where the calculation of the band width and of the shift is happening. Then these parameters are carried over into peptide alignment.